src/export/StructureDefinitionExporter.ts

```diff
@@ -544,7 +548,41 @@ export class StructureDefinitionExporter implements Fishable {
               rule.value = instance;
             }
             const replacedRule = replaceReferences(rule, this.tank, this);
-            element.assignValue(replacedRule.value, replacedRule.exactly, this);
+            try {
+              element.assignValue(replacedRule.value, replacedRule.exactly, this);
+            } catch (originalErr) {
+              // if an Instance has an id that looks like a number, bigint, or boolean,
+              // we may have tried to assign that value instead of an Instance.
+              // try to fish up an Instance with the rule's raw value.
+              // if we find one, try assigning that instead.
+              if (
+                originalErr instanceof MismatchedTypeError &&
+                (typeof rule.value === 'number' ||
+                  typeof rule.value === 'bigint' ||
+                  typeof rule.value === 'boolean')
+              ) {
+                const instanceExporter = new InstanceExporter(this.tank, this.pkg, this.fisher);
+                const instance = instanceExporter.fishForFHIR(rule.rawValue);
+                if (instance == null) {
+                  throw originalErr;
+                } else {
+                  try {
+                    element.assignValue(instance, rule.exactly, this);
+                    rule.value = instance;
+                  } catch (instanceErr) {
+                    // if it's still the wrong type, the assignment will fail.
+                    // but, we want to log that error with the original type, not the instance type.
+                    if (instanceErr instanceof MismatchedTypeError) {
+                      throw originalErr;
+                    } else {
+                      throw instanceErr;
+                    }
```

Comment:
Can we just always throw `originalErr` here? What type of exception could we get in this block (when we're _trying_ to assign an instance) for which we wouldn't want to just fall back to the original error anyway?

Reply:
Maybe not. I don't know. This is an honest question. If we're OK just always throwing `originalErr` when the inline instance assignment fails _for any reason_, then I think there are a bunch of `if`/`else` blocks in several files that can be simplified. So it's tempting... but only if we don't think there's much to lose.

Reply:
I was mostly considering `ValueAlreadyAssignedError` here, which would come up when the instance is the right type. I think in that case, it's more helpful to let the author know about the value conflict rather than an apparent type error which SUSHI has already managed to work through.

Reply:
Ah, that makes sense. I agree.